From e9216afdc4ab77eff472c5ba86792de1499e08ed Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas@debian.org>
Date: Thu, 11 Feb 2021 16:34:23 +0100
Subject: [PATCH 5/8] scripts/buildtools.mk: remove unneeded tests

The ?= had no effect when the previous test was succeeding.  Make that
explicit with an 'else'.

The 'ifdef' was always succeeding because previous stanza sets $1.
---
 scripts/mk/buildtools.mk | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/scripts/mk/buildtools.mk b/scripts/mk/buildtools.mk
index b2ab2a2ac..f05a9177e 100644
--- a/scripts/mk/buildtools.mk
+++ b/scripts/mk/buildtools.mk
@@ -37,16 +37,16 @@ define dpkg_buildtool_setvar
 ifeq (,$(findstring $(3),$(DEB_BUILD_OPTIONS)))
 ifeq ($(origin $(1)),default)
 $(1) = $(DEB_HOST_GNU_TYPE)-$(2)
-endif
+else
 $(1) ?= $(DEB_HOST_GNU_TYPE)-$(2)
+endif
 
 # On native build fallback to use TOOL if that's defined.
 ifeq ($(DEB_BUILD_GNU_TYPE),$(DEB_HOST_GNU_TYPE))
-ifdef $(1)
 $(1)_FOR_BUILD ?= $$($(1))
-endif
-endif
+else
 $(1)_FOR_BUILD ?= $(DEB_BUILD_GNU_TYPE)-$(2)
+endif
 else
 $(1) = :
 $(1)_FOR_BUILD = :
-- 
2.30.2

